ui/src/main/java/edu/wpi/grip/ui/codegeneration/CppTMethods.java

```diff
@@ -0,0 +1,51 @@
+package edu.wpi.grip.ui.codegeneration;
+
+import edu.wpi.grip.ui.codegeneration.data.TInput;
+import edu.wpi.grip.ui.codegeneration.data.TStep;
+
+import com.google.common.base.CaseFormat;
+
+import java.util.Locale;
+
+public class CppTMethods extends TemplateMethods {
+  public CppTMethods() {
+    super();
+  }
+
+  @Override
+  public String name(String name) {
+    Locale locName = new Locale(name);
+    return CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, locName.toString().toUpperCase());
+  }
+
+  @Override
+  public String callOp(TStep step) {
+    StringBuilder out = new StringBuilder();
+    if (step.name().equals("Switch") || step.name().equals("Valve")) {
+      out.append("pipeline");
+    }
+    out.append(name(step.name())).append('(');
+    for (TInput input : step.getInputs()) {
+      out.append(input.name());
+      out.append(", ");
+    }
+    if (step.name().equals("Threshold_Moving")) {
+      out.append("this->lastImage");
+      out.append(step.num());
+      out.append(", ");
```

Comment:
Can you do this differently? What if I add another operation that does something similar to Threshold Moving? I would have to change this other random line in the generator.